client/src/components/graphs/OrganizationalChart.js

```diff
@@ -69,15 +72,34 @@ const transitionDuration = 200
 
 const ranks = Settings.fields.person.ranks.map(rank => rank.value)
 
-const sortPositions = (positions, truncateLimit) => {
-  const allResults = [...positions].sort((p1, p2) =>
-    ranks.indexOf(p1.person?.rank) > ranks.indexOf(p2.person?.rank) ? -1 : 1
-  )
+const leaderCountMap = {}
+
+const sortPositions = (uuid, positions, truncateLimit) => {
+  const allResults = [...positions]
+    .sort((p1, p2) => {
+      if (p1.role === p2.role) {
+        return ranks.indexOf(p1.person?.rank) > ranks.indexOf(p2.person?.rank)
+          ? -1
+          : 1
+      }
+      return Object.keys(PositionRole).indexOf(p1.role) >
+        Object.keys(PositionRole).indexOf(p2.role)
+        ? -1
+        : 1
+    })
+    .map(pos => ({ ...pos, orgUuid: uuid }))
   return truncateLimit !== undefined && truncateLimit < allResults.length
     ? allResults.slice(0, truncateLimit)
     : allResults
 }
 
+const updateLeaderCount = (uuid, positions) => {
+  const count = positions.filter(
+    pos => pos.role === PositionRole.LEADER.toString()
+  ).length
+  leaderCountMap[uuid] = count > 0 ? count : 1
```

Comment:
It is unclear to me what this means. If there are zero leaders, the count is set to one?
[Side note: since array length is guaranteed to be non-negative, `count || 1` would have the same effect, which is a little DRY'er.]